src/WebJobs.Script.WebHost/Middleware/HostWarmupMiddleware.cs

```diff
@@ -19,8 +20,12 @@ public HostWarmupMiddleware(RequestDelegate next)
         {
             if (StandbyManager.IsWarmUpRequest(httpContext.Request))
             {
-                var scriptHostManager = (WebScriptHostManager)httpContext.Items[ScriptConstants.AzureFunctionsHostManagerKey];
-                await StandbyManager.WarmUp(httpContext.Request, scriptHostManager);
+                if (!httpContext.Items.TryGetValue(ScriptConstants.AzureFunctionsHostManagerKey, out object scriptHostManager))
+                {
+                    throw new InvalidOperationException($"Warmup request received, but no instance of {nameof(WebScriptHostManager)} in HTTP context");
```

Comment:
When would this happen? If we wait for the host to start up won't it be set? Who sets this key?

Reply:
This is just a safety check as the current process (with the middleware order and setup) is a bit fragile and something I intend to address in a future PR. Having this in place would just make troubleshooting simpler if it ever occurs.